app/Http/Controllers/PollerGroupController.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Device;
+use App\Models\Poller;
+use App\Models\PollerGroups;
+use App\Models\PollerCluster;
+use Illuminate\Validation\Rule;
+use Illuminate\Http\Request;
+use LibreNMS\Alerting\QueryBuilderFilter;
+use LibreNMS\Alerting\QueryBuilderFluentParser;
+use LibreNMS\Config;
+use Toastr;
+
+class PollerGroupController extends Controller
+{
+    public $rrdstep;
+    public $isDistributedPoller;
+    public $defaultPollerId;
+
+    public $defaultPollerTab = 'poller';
+
+    public $defaultGroup = ['id' => 0,
+                             'group_name' => 'General',
+                             'descr' => ''];
+    public $defaultPollerMarker = '(default Poller)';
+
+    public function __construct()
+    {
+        $this->authorizeResource(PollerGroup::class, 'poller_groups');
+
+        $this->rrdstep = \LibreNMS\Config::get('rrd.step');
+
+        $this->isDistributedPoller = \LibreNMS\Config::get('distributed_poller');
+        $this->defaultPollerId = \LibreNMS\Config::get('distributed_poller_group');
+    }
+
+    public function index(Request $request)
+    {
+        $current_tab = $request['tab'] ?: $this->defaultPollerTab;
+
+        switch ($current_tab) {
```

Comment:
This switch statement is messy, I'd prefere explicit routes that point to a specific function for each tab.
Also, you can inject the tab int the function `public function index(Request $request, $tag)`

Reply:
don't see what you mean.
if no tab is defined -> default tab
if defined tab not matching -> fall back to default tab

i can change it to a ifelse chain if you'd like to, but it does the same and will not look much nicer in my eyes ;)